crates/modelardb_manager/src/metadata.rs

```diff
@@ -13,188 +13,478 @@
-
-            Ok(manager_key
-                .parse()
-                .map_err(|error| sqlx::Error::ColumnDecode {
-                    index: "key".to_owned(),
-                    source: Box::new(error),
-                })?)
-        } else {
+        let keys = array!(batch, 0, StringArray);
+        if keys.is_empty() {
             let manager_key = Uuid::new_v4();
 
             // Add a new row to the manager_metadata table to persist the key.
-            sqlx::query("INSERT INTO manager_metadata (key) VALUES ($1)")
-                .bind(manager_key.to_string())
-                .execute(&self.metadata_database_pool)
+            self.metadata_delta_lake
+                .append_to_table(
+                    "manager_metadata",
+                    vec![Arc::new(StringArray::from(vec![manager_key.to_string()]))],
+                )
                 .await?;
 
             Ok(manager_key)
+        } else {
+            let manager_key: String = keys.value(0).to_owned();
+
+            Ok(manager_key
+                .parse()
+                .map_err(|error: uuid::Error| DeltaTableError::Generic(error.to_string()))?)
         }
     }
 
-    /// Save the node to the metadata database and return [`Ok`]. If the node could not be saved,
-    /// return [`sqlx::Error`].
-    pub async fn save_node(&self, node: &Node) -> Result<(), sqlx::Error> {
-        sqlx::query("INSERT INTO nodes (url, mode) VALUES ($1, $2)")
-            .bind(&node.url)
-            .bind(node.mode.to_string())
-            .execute(&self.metadata_database_pool)
+    /// Save the node to the metadata Delta Lake and return [`Ok`]. If the node could not be saved,
+    /// return [`DeltaTableError`].
+    pub async fn save_node(&self, node: &Node) -> Result<(), DeltaTableError> {
+        self.metadata_delta_lake
+            .append_to_table(
+                "nodes",
+                vec![
+                    Arc::new(StringArray::from(vec![node.url.clone()])),
+                    Arc::new(StringArray::from(vec![node.mode.to_string()])),
+                ],
+            )
             .await?;
 
         Ok(())
     }
 
-    /// Remove the row in the nodes table that corresponds to the node with `url` and return [`Ok`].
-    /// If the row could not be removed, return [`sqlx::Error`].
-    pub async fn remove_node(&self, url: &str) -> Result<(), sqlx::Error> {
-        sqlx::query("DELETE FROM nodes WHERE url = $1")
-            .bind(url)
-            .execute(&self.metadata_database_pool)
+    /// Remove the row in the `nodes` table that corresponds to the node with `url` and return
+    /// [`Ok`]. If the row could not be removed, return [`DeltaTableError`].
+    pub async fn remove_node(&self, url: &str) -> Result<(), DeltaTableError> {
+        let ops = self
+            .metadata_delta_lake
+            .metadata_table_delta_ops("nodes")
             .await?;
 
+        ops.delete().with_predicate(col("url").eq(lit(url))).await?;
```

Comment:
Maybe add a comment why Apache DataFusion is not used despite being used below.

Reply:
I have not added a comment since I think it makes sense that we use `DeltaOps` whenever possible. This will probably also become more self-explanatory when `MetadataDeltaLake` is removed in the future to get rid of a lot of the "ad hoc" code.